data/src/main/java/me/androidbox/data/worker_manager/FullAgendaSynchronizerImp.kt

```diff
@@ -0,0 +1,45 @@
+package me.androidbox.data.worker_manager
+
+import androidx.work.BackoffPolicy
+import androidx.work.Constraints
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkRequest
+import me.androidbox.domain.work_manager.FullAgendaSynchronizer
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class FullAgendaSynchronizerImp @Inject constructor(
+    private val workManager: WorkManager) : FullAgendaSynchronizer {
+
+    companion object {
+        const val THIRTY_MINUTES = 30L
+        const val FULL_AGENDA_SYNC = "full_agenda_sync"
+    }
+
+    override suspend fun sync(): UUID {
+        val fullSynWorkerRequest = PeriodicWorkRequestBuilder<SyncFullAgendaItemsWorker>(
+            repeatInterval = THIRTY_MINUTES,
+            repeatIntervalTimeUnit = TimeUnit.MINUTES)
+            .setConstraints(Constraints(Constraints.Builder().setRequiredNetworkType(NetworkType.CONNECTED).build()))
+            .setBackoffCriteria(
+                BackoffPolicy.EXPONENTIAL,
+                WorkRequest.MIN_BACKOFF_MILLIS,
+                TimeUnit.MILLISECONDS
+            ).build()
+
+        workManager.enqueueUniquePeriodicWork(
+            FULL_AGENDA_SYNC,
+            ExistingPeriodicWorkPolicy.CANCEL_AND_REENQUEUE,
+            fullSynWorkerRequest)
+
+        return fullSynWorkerRequest.id
```

Comment:
Just wondering, what is the purpose of returning a UUID of the worker if I am not using the result anywhere?

Reply:
Returning the ID is used if in the future you want to observe the state

On tasky, it's not really necessary since you won't use it